From ad0e85663d22632ace82185e36a9eb6828b1ee35 Mon Sep 17 00:00:00 2001
From: Yang Xun <yangxun@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:05 +0800
Subject: [PATCH 061/150] dt-bindings: PCI: phytium: Add DT bindings for PCIe
 endpoint controller

This patch documents the DT bindings for the Phytium PCIe controller
when configured in endpoint mode.

Signed-off-by: Yang Xun <yangxun@phytium.com.cn>
Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: I3cf09015575ebec74dd999cd0233bee55e529c33
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../bindings/pci/phytium,pd2008-pcie-ep.yaml  | 42 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 43 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/pci/phytium,pd2008-pcie-ep.yaml

diff --git a/Documentation/devicetree/bindings/pci/phytium,pd2008-pcie-ep.yaml b/Documentation/devicetree/bindings/pci/phytium,pd2008-pcie-ep.yaml
new file mode 100644
index 000000000000..734c197b3983
--- /dev/null
+++ b/Documentation/devicetree/bindings/pci/phytium,pd2008-pcie-ep.yaml
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/pci/phytium,pd2008-pcie-ep.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium PCIe endpoint controller
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+allOf:
+  - $ref: "pci-ep.yaml#"
+
+properties:
+  compatible:
+    const: phytium,pd2008-pcie-ep
+
+  reg:
+    maxItems: 2
+
+  reg-names:
+    items:
+      - const: reg
+      - const: mem
+
+required:
+  - compatible
+  - reg
+  - reg-names
+
+examples:
+  - |
+    ep0: ep@0x29030000 {
+      compatible = "phytium,pd2008-pcie-ep";
+      reg = <0x0 0x29030000 0x0 0x10000>,
+            <0x11 0x00000000 0x1 0x00000000>,
+            <0x0 0x29101000 0x0 0x1000>;
+      reg-names = "reg", "mem", "hpb";
+      max-outbound-regions = <3>;
+      max-functions = /bits/ 8 <1>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 43988de6cc67..cf96694ff0a6 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2546,6 +2546,7 @@ F:	Documentation/devicetree/bindings/mmc/phytium,mci.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,sdci.yaml
 F:	Documentation/devicetree/bindings/mtd/phytium,nfc.yaml
 F:	Documentation/devicetree/bindings/net/can/phytium,can.yaml
+F:	Documentation/devicetree/bindings/pci/phytium,pd2008-pcie-ep.yaml
 F:	Documentation/devicetree/bindings/pwm/phytium,pwm.yaml
 F:	Documentation/devicetree/bindings/rng/phytium,rng.yaml
 F:	Documentation/devicetree/bindings/sound/phytium,hda.yaml
-- 
2.47.0

